src/TplStats.Core/Models/Game.cs

```diff
@@ -0,0 +1,153 @@
+        public static implicit operator GenProp<T>(T value) {
+            return new GenProp<T> { Value = value };
+        }
+    }
+    public class Game : IEntity {
+        // Generic version of property settings.  Each of the following should
+        // be automatically screened for null values and other sanitization
+        // on get and set operations.
+        private GenProp<string> _name             { get; set; }
+        private GenProp<string> _field            { get; set; }
+        private GenProp<Team> _homeTeam           { get; set; }
+        private GenProp<Team> _awayTeam           { get; set; }
+        // start and end times have additional checking, so it's better
+        // to use non-generic properties.
+        private LocalDateTime _startTime;
+        private LocalDateTime _endTime;
+        // This is the non-generic version -- keep it around until we know
+        // the generic version works properly.
+        // The reason I wanted to switch over, was just for the sake of cleaning
+        // up code duplication -MG.
+        /*
+        private string _name;
+        private string _field;
+        private LocalDateTime _startTime;
+        private LocalDateTime _endTime;
+        private Team _homeTeam;
+        private Team _awayTeam;
+
+        public int Id { get; private set; }
+        public string Name {
+            get => _name;
+            set {
+                if (value is null) {
+                    throw new ArgumentNullException
+                        (nameof(value),"Name cannot be null");
+                }
+                _name = value.Trim();
+            }
+        }
+        public string Field {
+            get => _field;
+            set {
+                if (value is null) {
+                    throw new ArgumentNullException
+                        (nameof(value),"Field cannot be null");
+                }
+                _field = value.Trim();
+            }
+        }
+        public Team HomeTeam {
+            get => _homeTeam;
+            set {
+                if (value is null) {
+                    throw new ArgumentNullException
+                        (nameof(value),"cannot be null");
+                }
+                _homeTeam = value;
+            }
+        }
+        public Team AwayTeam {
+            get => _awayTeam;
+            set {
+                if (value is null) {
+                    throw new ArgumentNullException
+                        (nameof(value),"cannot be null");
+                }
+                _awayTeam = value;
+            }
+        }
+        */
+        public LocalDateTime StartTime {
+            get => _startTime;
+            set {
+                if (value <= EndTime) {
+                    _startTime = value;
+                }
+                else {
+                    throw new ArgumentOutOfRangeException(
+                        nameof(value),
+                        $"cannot be after end time ({EndTime})");
```

Comment:
`ArgumentOutOfRangeException`has a [constructor](https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception.-ctor?view=netframework-4.7.2#System_ArgumentOutOfRangeException__ctor_System_String_System_Object_System_String_) that allows you to pass the argument's value. (I made this same mistake in my code - I'll have to fix that.)

```suggestion
                        value,
                        $"cannot be after end time ({EndTime})");
```